src/TurbulenceAveragingPostProcessing.C

```diff
@@ -735,6 +869,233 @@ TurbulenceAveragingPostProcessing::compute_favre_stress(
+          const int component = componentCount;
+          const double uj = uNp1[k*nDim+j];
+          const double newStress
+            = (stress[k*stressSize+component]*oldTimeFilter*zeroCurrent
+               + rho*ui*uj*dt)/currentTimeFilter_ ;
+          stress[k*stressSize+component] = newStress;
+          componentCount++;
+        }
+      }
+    }
+  }
+}
+
+//--------------------------------------------------------------------------
+//-------- compute_vortictiy -----------------------------------------------
+//--------------------------------------------------------------------------
+void
+TurbulenceAveragingPostProcessing::compute_sfs_stress(
+  const std::string &averageBlockName,
+  const double &oldTimeFilter,
+  const double &zeroCurrent,
+  const double &dt,
+  stk::mesh::Selector s_all_nodes)
+{
+  stk::mesh::MetaData & metaData = realm_.meta_data();
+
+  const int nDim = realm_.spatialDimension_;
+  const double invNdim = 1.0/nDim;
+
+  const std::string SFSStressFieldName = "sfs_stress";
+
+  bool computeSFSTKE = false;
+  // extract fields
+  stk::mesh::FieldBase *TurbViscosity_ = metaData.get_field(stk::topology::NODE_RANK, "turbulent_viscosity");
+  stk::mesh::FieldBase *TurbKe_ = metaData.get_field(stk::topology::NODE_RANK, "turbulent_ke");
+  if(TurbKe_ == NULL) computeSFSTKE = true ;
+  stk::mesh::FieldBase *DualNodalVolume_ = metaData.get_field(stk::topology::NODE_RANK, "dual_nodal_volume");
+  stk::mesh::FieldBase *DuDx_ = metaData.get_field(stk::topology::NODE_RANK, "dudx");
+  stk::mesh::FieldBase *SFSStress = metaData.get_field(stk::topology::NODE_RANK, SFSStressFieldName);
+
+  stk::mesh::BucketVector const& node_buckets_sfsstress =
+    realm_.get_buckets( stk::topology::NODE_RANK, s_all_nodes );
+  for ( stk::mesh::BucketVector::const_iterator ib = node_buckets_sfsstress.begin();
+        ib != node_buckets_sfsstress.end() ; ++ib ) {
+    stk::mesh::Bucket & b = **ib ;
+    const stk::mesh::Bucket::size_type length   = b.size();
+
+    // fields
+    const double * turbNu_ = (double*)stk::mesh::field_data(*TurbViscosity_, b);
+    double * turbKe_;
+    if (!computeSFSTKE)
+        turbKe_ = (double*)stk::mesh::field_data(*TurbKe_, b);
+    const double * dualNodalVolume_ = (double*)stk::mesh::field_data(*DualNodalVolume_, b);
+    const double * dudx_ = (double*)stk::mesh::field_data(*DuDx_, b);
+    double *sfsstress_ = (double*)stk::mesh::field_data(*SFSStress,b);
+    const int offSet = 6;
+
+    // Store the xx, xy, xz, yy, yz, zz components of sfs_stress
+    for ( stk::mesh::Bucket::size_type k = 0 ; k < length ; ++k ) {
+      double divU = 0.0;
+      for ( int j = 0; j < nDim; ++j)
+          divU += dudx_[k*offSet+(nDim*j+j)] ;
+      double sfstke = 0.0;
+      if(computeSFSTKE) {
+          // Use method of Yoshisawa (1986) - Statistical theory for compressible turbulent shear flows, with the application to subgrid modeling, 29, 2152.
+          double Ci = 0.9;
```

Comment:
Probably shouldn't just be a hard coded value.  One option would be to just put Ci into the turbulent constants map.  Another would be to compute an estimate of Ci from the values of cmuCs and cmuEps already in the turbulent constants map ( ci = cs^4/cnu^2 ).  I'd prefer just putting Ci into the constants map.

Reply:
Thanks @rcknaus. I moved this to the turbulent constants map.